### DISCUSSION TITLE
Question: regexp `include` option as an alternative to `glob`

### DISCUSSION THREAD

**Discussion by noahp** [evilmartians/lefthook#710] [Ideas]
Love this tool, thanks for making it!

I've found the `glob` option to be a little restrictive, and would love a regexp `include` option (similar but inverse of the `exclude` option) that pattern-matches against the full relative path for files, if it makes sense to add!

EDIT-

Actually, using negative look-ahead with `exclude` might just do it for my use cases!

**Comment by mrexox** [comment]
Hey! Thank you for this suggestion. Indeed this is a good idea. Or maybe it can be named `regex` (to pair with `glob` option). I will have this feature on my radar but if you want to submit a PR - you are welcome!

**Comment by anthony-hayes** [comment]
I'm also finding the `glob` to be a little restrictive - an `include` list of globs would be very useful for me, as I have multiple globs to match (monorepo setup with incremental adoption of a certain pre-commit tool)

I have workarounds, but a list of globs would be ideal
